keanu-project/src/main/java/io/improbable/keanu/vertices/generic/probabilistic/discrete/CategoricalVertex.java

```diff
@@ -23,6 +25,22 @@
         return new CategoricalVertex<>(defensiveCopy(selectableValues));
     }
 
+    public static <T> CategoricalVertex<T> of(DirichletVertex vertex, List<T> categories) {
+        final int length = ArrayUtil.prod(vertex.getShape());
+        if (length != categories.size()) {
+            throw new IllegalArgumentException("Categories must have length of vertex's size");
+        }
+
+        final int categoriesCount = categories.size();
+
+        return new CategoricalVertex<>(IntStream.range(0, categoriesCount).boxed().collect(toMap(categories::get, i -> new TakeVertex(vertex, 0, i))));
```

Comment:
for clarity could you separate this into more than one line? in particular can you think of good variable names to explain some of the following steps:

    v1 = IntStream.range(0, categoriseCount).boxed()
    v2 = i -> new TakeVertex(vertex, 0, i)
    v3 = toMap(categories::get, v2)
    v4 = v1.collect(v3));

Reply:
Good call for splitting this and the other section up for readability. I'll try and strike a balance so as to not write so much that it loses readability.